src/python/turicreate/data_structures/sframe.py

```diff
@@ -1603,13 +1603,18 @@ def read_json(cls,
         [3 rows x 1 columns]
         """
         if orient == "records":
-            g = SArray.read_json(url)
-            if len(g) == 0:
-                return SFrame()
-            if g.dtype != dict:
-                raise RuntimeError("Invalid input JSON format. Expected list of dictionaries")
-            g = SFrame({'X1':g})
-            return g.unpack('X1','')
+            if type(url)==str and url[0] !='{' :
+                g = SArray.read_json(url)
+                if len(g) == 0:
+                    return SFrame()
+                if g.dtype != dict:
+                    raise RuntimeError("Invalid input JSON format. Expected list of dictionaries")
+                g = SFrame({'X1':g})
+                return g.unpack('X1','')
+            elif type(url)==str and url[0] =='{' :
+                url=pandas.read_json(url)
```

Comment:
actually we don't want to introduce pandas if it's not necessary due to the reason that pandas import time is lengthy (~260ms) and duplicated with SFrame in terms of functionality. Could you use json lib to read the json as a dictionary (you can lazily import here)? Then you can pass the dictionary to SFrame constructor.

Reply:
@Jarvi-Izana - Even without this change `pandas` has already been loaded. So I don't think we should worry about that load time in this pull request.
The reason `pandas` is being used here is to prevent having to write the JSON string to disk when creating a `SFrame`. I don't think anything in the standard `json` lib is going to be able to help with that.

Reply:
nope, the lazy module deferred this, if someday we'd like put that back. In general, we shouldn't eagerly load pandas since we encourage people to use sframe.

Reply:
I'm confused. Nothing in this pull request changes when `pandas` is loaded/imported. It's only adding an additional usage of `pandas` when a user calls `SFrame.read_json`.

Reply:
That's right. The lazy loading in the future should handle it.